File: player.py
import sys
import pdb
import time
data = sys.stdin.readline().strip()
moveFile = None
player = -1
allMoves = None

if(data == 'Player 1'):
	player = 1
	moveFile = open('whiteMoves.txt','rb')
	allMoves = moveFile.readlines()
else:
	player = 2
	moveFile = open('blackMoves.txt','rb')
	allMoves = moveFile.readlines()
if(player == 1):
	idx = 0
	while True:
		time.sleep(1)		
		move = allMoves[idx].strip()
		# Write move to client		
		move = move + '\n'		
		sys.stdout.write(move)
		sys.stdout.flush()
		# read move from client
		move = sys.stdin.readline()
		idx += 1
		if(idx == len(allMoves)):
			break
elif(player == 2):
	idx = 0
	while True:		
		# read move from client
		move = sys.stdin.readline()
		time.sleep(1)
		move = allMoves[idx].strip()
		#Send move from client
		sys.stdout.write(move + "\n")
		sys.stdout.flush()
		idx += 1
		if(idx == len(allMoves)):
			break

File: player.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm> 
#include <functional> 
#include <cctype>
#include <locale>
using namespace std;
// trim from start
static inline std::string &ltrim(std::string &s) {
    s.erase(s.begin(), std::find_if(s.begin(), s.end(),
            std::not1(std::ptr_fun<int, int>(std::isspace))));
    return s;

}

// trim from end
static inline std::string &rtrim(std::string &s) {
    s.erase(std::find_if(s.rbegin(), s.rend(),
            std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
    return s;

}

// trim from both ends
static inline std::string &trim(std::string &s) {
    return ltrim(rtrim(s));

}
void getAllMoves(vector<string>& allMoves, string fileName){
	ifstream moveFile;
	moveFile.open(fileName);
	while(!moveFile.eof()){
		string move;
		moveFile >> move;
		allMoves.push_back(move);
	}
}
int main(){
	string data;
	int n,TL;
	cin>>data>>n>>TL;
	int player = -1;
	vector<string> allMoves;
	if(data == "1"){
		player = 1;
		getAllMoves(allMoves, "whiteMoves.txt");
	}
	else{
		player = 2;
		getAllMoves(allMoves, "blackMoves.txt");	
	}
	if(player == 1){
		int idx = 0;
		while(1){
			string move = trim(allMoves[idx]);
			move += '\n';
			cout<<move;
			cin>>move;
			idx += 1;
			if( idx == allMoves.size() )
				break;			
		}
	}
	else{
		int idx = 0;
		while(1){
			string move;
			cin >> move;
			move = trim(allMoves[idx]);
			cout<<move<<"\n";
			idx ++;
			if(idx == allMoves.size()){
				break;
			}		
		}
	}

}

